agent/check_handler_internal_test.go

```diff
@@ -43,13 +44,13 @@ func TestHandleCheck(t *testing.T) {
 	agent.inProgressMu.Lock()
 	agent.inProgress[checkKey(request)] = request.Config
 	agent.inProgressMu.Unlock()
-	assert.Error(agent.handleCheck(payload))
+	assert.Error(agent.handleCheck(context.TODO(), payload))
```

Comment:
Just curious about the choice of `context.TODO()` over `context.Background()`. They boil down to the same context, but I would have gone with `Background()` here.

Reply:
Yeah my interpretation of the docs was that `context.Background()` is typically intended for tests (or the main function). They both return the same empty context, so its definitely a small thing!

Reply:
My thinking at the time was that we might want to cancel the context when the test terminates one day.